pkg/controller/association/conf.go

```diff
@@ -173,10 +175,47 @@ func extractAssociationConf(annotations map[string]string, annotationName string
 	return &assocConf, nil
 }
 
+func RemoveExcesiveAssociationConfs(
+	client k8s.Client,
+	associated commonv1.Associated,
+	associations []commonv1.Association,
+	associationConfAnnotationNameBase string,
+) error {
+	accessor := meta.NewAccessor()
+	annotations, err := accessor.Annotations(associated)
+	if err != nil {
+		return err
+	}
+
+	expected := make(map[string]bool)
+	for _, association := range associations {
+		key := commonv1.FormatNameWithID(associationConfAnnotationNameBase+"%s", association.ID())
+		expected[key] = true
+	}
+
+	modified := false
+	for key := range annotations {
+		if strings.HasPrefix(key, associationConfAnnotationNameBase) && !expected[key] {
+			delete(annotations, key)
+			modified = true
+		}
+	}
+
+	if !modified {
+		return nil
+	}
+
+	if err := accessor.SetAnnotations(associated, annotations); err != nil {
+		return err
+	}
+
+	return client.Update(associated)
+}
+
 // RemoveAssociationConf removes the association configuration annotation.
-func RemoveAssociationConf(client k8s.Client, obj runtime.Object, annotationName string) error {
+func RemoveAssociationConf(client k8s.Client, associated commonv1.Associated, annotationNameBase string, id int) error {
```

Comment:
Could this not simply be 
```Golang
func RemoveAssociationConf(client k8s.Client, association commonv1.Association) error {
```
assuming you could something like to derive the annotation name?

```golang

	annotationName := association.ID().AnnotationName()
```